### PR TITLE
Added iOS testing tools cheat sheet

### PR DESCRIPTION
**Pull request by mish24**
## Description of new Instant Answer, or changes

This cheat sheet provides a checklist of tasks to be performed when testing an iOS application. When assessing a mobile application several areas should be taken into account: client software, the communication channel and the server side infrastructure. Testing an iOS application usually requires a jailbroken device, which is basically a device that not pose any restrictions on the software that can be installed on it. 

## People to notify

@kirkins 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/ios_testing_tools

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by mish24** [comment]
@kirkins Have a look please

File: share/goodie/cheat_sheets/json/iostesting-tools.json [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                "key": "Mallory proxy",
+                "val": "Proxy for Binary protocols"
+            },
+            {
+                "key": "Charles/Burp proxy",
+                "val": "Proxy for HTTP and HTTPS"
+            },
+            {
+                "key": "OpenSSH",
+                "val": "Connect to the iPhone remotely over SSH",
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/iostesting-tools.json]: Remove the extra `, ` at end of this line

File: share/goodie/cheat_sheets/json/iostesting-tools.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            },
+            {
+                "key": "Sqlite3",
+                "val": " 	Sqlite database client"
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/iostesting-tools.json]: Remove the big tab here.

**Comment by mish24** [comment]
@kirkins Thank you! That is really awesome. I checked in that and now atleast the json is validated.

**Comment by mish24** [comment]
@kirkins Passed. Thanks again :D

File: share/goodie/cheat_sheets/json/iostesting-tools.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
+{
+    "id": "iostesting_tools_cheat_sheet",
+    "name": "IOSTestingTools",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/iostesting-tools.json]: Lets add space between words

File: share/goodie/cheat_sheets/json/iostesting-tools.json [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    "description": "This cheat sheet provides a checklist of tasks to be performed when testing an iOS application. When assessing a mobile application several areas should be taken into account: client software, the communication channel and the server side infrastructure. Testing an iOS application usually requires a jailbroken device, which is basically a device that not pose any restrictions on the software that can be installed on it.",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/iostesting-tools.json]: Would be better to have a short description about the content cheat sheet

File: share/goodie/cheat_sheets/json/iostesting-tools.json [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                "key": "BinaryCookieReader",
+                "val": "Read cookies.binarycookies files"
+            },
+            {
+                "key": "lsof ARM Binary",
+                "val": "list of all open files and the processes that opened them"
+            },
+            {
+                "key": "lsock ARM Binary",
+                "val": "monitor socket connections"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/iostesting-tools.json]: Start the val field with capital letter.  
Please do check val fields above and below this comment for the same issue.

**Comment by mish24** [comment]
@jophab Alright! Thanks for helping out. I'll update the PR as soon as i can :)

File: share/goodie/cheat_sheets/json/iostesting-tools.json [hunk 6]
```diff
@@ -0,0 +1,148 @@
+{
+    "id": "iostesting_tools_cheat_sheet",
+    "name": "IOSTestingTools",
+    "description": "This cheat sheet provides a checklist of tasks to be performed when testing an iOS application. When assessing a mobile application several areas should be taken into account: client software, the communication channel and the server side infrastructure. Testing an iOS application usually requires a jailbroken device, which is basically a device that not pose any restrictions on the software that can be installed on it.",
+    "metadata": {
+        "sourceName": "A list of valuable tools for testing iOS devices."
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/iostesting-tools.json]: Can you add the source url you referred to make this cheat sheet as sourceUrl and update the sourceName accordingly to represent the domain name of source url.
Like "Wikipedia" for "www.wikipedia.org/"

Comment (mish24) [inline share/goodie/cheat_sheets/json/iostesting-tools.json]: Okay. It's mixed from many resources, so I'll add the one which I referred to the most

Comment (jophab) [inline share/goodie/cheat_sheets/json/iostesting-tools.json]: Yeah.. Thats fine :+1:

**Comment by mish24** [comment]
@jophab Hi! I updated my PR with the suggested improvements. Please have a look :D

File: share/goodie/cheat_sheets/json/iostesting-tools.json [hunk 7]
```diff
@@ -1,9 +1,10 @@
 {
     "id": "iostesting_tools_cheat_sheet",
-    "name": "IOSTestingTools",
-    "description": "This cheat sheet provides a checklist of tasks to be performed when testing an iOS application. When assessing a mobile application several areas should be taken into account: client software, the communication channel and the server side infrastructure. Testing an iOS application usually requires a jailbroken device, which is basically a device that not pose any restrictions on the software that can be installed on it.",
+    "name": "IOS Testing Tools",
+    "description": "This cheat sheet provides a list of tools and frameworks we can use for testing iOS devices",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/iostesting-tools.json]: "A list of tools and frameworks we can use for testing iOS devices"

Would be enough here

File: share/goodie/cheat_sheets/json/iostesting-tools.json [hunk 8]
```diff
@@ -1,9 +1,10 @@
...
     "metadata": {
-        "sourceName": "A list of valuable tools for testing iOS devices."
+        "sourceUrl": "www.owasp.org/index.php/IOS_Application_Security_Testing",
+        "sourceName": "OWASP Application Testing references"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/iostesting-tools.json]: Change SourceName to "OWASP"

**Comment by mish24** [comment]
@jophab I have made the required changes in both PRs

**Comment by sahildua2305** [comment]
@mish24 the [sourceURL](https://www.owasp.org/index.php/IOS_Application_Security_Testing) doesn't contain anything.

**Comment by mish24** [comment]
@sahildua2305 Will changing the name in the json field resolve this?

**Comment by mintsoft** [comment]
@mish24 Are you ok with the changes @sahildua2305 has raised here?

**Comment by mish24** [comment]
@mintsoft Yes! I'll fix it soon.

**Comment by mish24** [comment]
@sahildua2305 the name comes out to be `ios-testing-tools.json` by default, and not with underscores. :(

**Comment by kirkins** [comment]
@mish24 you have 2 PR for the same cheat sheet right now.

Please close one of them.

**Pull request closed without merging**